src/mono/mono/mini/mini-runtime.c

```diff
@@ -143,8 +143,8 @@ static mono_mutex_t jit_mutex;
 
 static MonoCodeManager *global_codeman;
 
-MonoDebugOptions mini_debug_options;
-char *sdb_options;
+MONO_COMPONENT_API MonoDebugOptions mini_debug_options;
+MONO_COMPONENT_API char *sdb_options;
```

Comment:
Is there a cheap way to avoid making data public?

1. I think there's some complication with exporting data from DLLs on Windows, isn't there?
2. we have `MONO_API_DATA` in addition to `MONO_API` which is there to export data symbols.

I think it's better to just give the runtime more component methods to populate this stuff.  or give the component a new runtime function that it can call to access the data.

If you need to read the data fast in the runtime, but set it in the component, take a look at what I did in hot_reload `MonoMetadataUpdateData` https://github.com/dotnet/runtime/search?q=MonoMetadataUpdateData  Add a struct in the runtime to hold all the data that you want to access quickly.  Then add a method to the component like `set_fastpath_data` and call it early in mono startup to initialize to give the component a pointer to the variable that lives in the runtime.  Then the component sets the data through the pointer, and the runtime reads the data through its global variable.

Reply:
done.